src/Funogram.Telegram/Bot.fs

```diff
@@ -1,97 +1,149 @@
+            processUpdates updates
+            return! loopAsync offset // send new offset
+          | Error e ->
+            config.OnError (e.AsException() :> Exception)
             return! loopAsync offset
-        }
-    loopAsync (config.Offset |> Option.defaultValue 0L)
+          | _ -> return! loopAsync offset
+        with ex ->
+          config.OnError ex
+          return! loopAsync (offset + 1L)
+        return! loopAsync offset
+      }
 
-let startBot config updateArrived updatesArrived =
+    loopAsync (config.Offset |> Option.defaultValue 0L)
+  | Some webHook ->
     async {
-        let! me = getMe |> api config
-        return! me 
-        |> function
-        | Error error -> failwith error.Description
-        | Ok me -> runBot config me updateArrived updatesArrived
+      let listener = webHook.Listener
+      let validateRequest = webHook.ValidateRequest
+
+      if not listener.IsListening then listener.Start()
+
+      while listener.IsListening do
+        let! context = listener.GetContextAsync() |> Async.AwaitTask
+        
+        if validateRequest context.Request then
+          match Funogram.Tools.parseJsonStream<Update> context.Request.InputStream with
+          | Ok updates -> processUpdates (seq { updates })
+          | Error e -> config.OnError e
+        else
+          context.Response.StatusCode <- 403
+        context.Response.Close()
+          
     }
+    
+let startBot config updateArrived updatesArrived =
+  async {
+    let! me = getMe |> api config
+    return! me 
+    |> function
+    | Error error -> failwith error.Description
+    | Ok me -> runBot config me updateArrived updatesArrived
+  }
 
 let processCommands (context: UpdateContext) =
-    Seq.forall (fun command -> command context)
+  Seq.forall (fun command -> command context)
+
+(*
+async {
+     if not listener.IsListening then listener.Start()
+
+     while listener.IsListening do
+       let! context = listener.GetContextAsync() |> Async.AwaitTask
+       if context.Request.IsWebSocketRequest then
+         let! wsContext = context.AcceptWebSocketAsync(null) |> Async.AwaitTask
+         use ws = wsContext.WebSocket
+         
+         let buffer = ArraySegment<byte>(Array.zeroCreate 1024)
+         use ms = new MemoryStream()
+
+         while ws.State = WebSocketState.Open do
+           let! result = ws.ReceiveAsync(buffer, Async.DefaultCancellationToken) |> Async.AwaitTask
+           do! ms.AsyncWrite(buffer.Array, buffer.Offset, buffer.Count)
+
+           if result.EndOfMessage then
+             match Funogram.Tools.parseJsonStream<Update seq> ms with
+             | Ok updates -> processUpdates updates
+             | Error e -> config.OnError e
+             ms.Flush()
+       else
+         match Funogram.Tools.parseJsonStream<Update> context.Request.InputStream with
+         | Ok updates -> processUpdates (seq { updates })
+         | Error e -> config.OnError e
+         
+   }
+*)
```

Comment:
Why is this commented out code here? Is it still useful for some debug purposes? Then, at least please state the purpose in the comments, or just delete it :)